test/blackbox-tests/test-cases/foreign-objects-lib.t

```diff
@@ -0,0 +1,106 @@
+----------------------------------------------------------------------------------
+Test suite for (foreign_objects ...) field.
```

Comment:
I haven't looked at the tests too closely, but if it's at all possible, it's better to split individual tests into separate .t files. This helps keeps the test cases more isolated and allows for more tests to run concurrently. This is especially important for tests that run `ocamlc` or `cc`

Reply:
I've split up `foreign-objects-lib.t` so that the test where the object files are generated by rules is in a separate file. Other than that, I think each file is testing a single thing. Let me know if you think they should be more fine grained.